openhcl/build_info/src/lib.rs

```diff
@@ -60,9 +60,11 @@ impl BuildInfo {
 // a debugger. With a debugger, the non-mangled name is easier
 // to use.
 
-// UNSAFETY: link_section and export_name are considered unsafe.
+// UNSAFETY: link_section and export_name are unsafe.
 #[expect(unsafe_code)]
+// SAFETY: The build_info section is custom and carries no safety requirements.
 #[unsafe(link_section = ".build_info")]
+// SAFETY: The name "BUILD_INFO" does not collide with any other symbols.
```

Comment:
I mean, _technically speaking_, there's nothing stopping some other crate in our build graph from exporting the same symbol name / taking an `extern` dep on this symbol existing. So I don't think this particular phrasing is accurate / resilient.

I would tweak this wording with more justification discussing the specifics of how this crate is used in OpenHCL, and touching on the fact that since we control our dep graph here, it shouldn't be an issue.

* * *

Though, honestly, it might be better if this crate was reworked to export a macro that the top-level binary crate could invoke, vs. having this static get declared in a sub-crate...